examples/Azure_IoT_Hub_Arduino_Nano_RP2040_Connect/Azure_IoT_Hub_Arduino_Nano_RP2040_Connect.ino

```diff
@@ -0,0 +1,591 @@
+
+  mqttClient.beginMessage(telemetryTopic);
+  mqttClient.print(generateTelemetry());
+  mqttClient.endMessage();
+
+  LogInfo("Telemetry sent.");
+  delay(100);
+  digitalWrite(LED_PIN, LOW);
+}
+
+/*
+ * generateTelemetry:
+ * Simulated telemetry.  
+ * In your application, this function should retrieve real telemetry data from the device and format
+ * it as needed.
+ */
+static char* generateTelemetry() 
+{
+  String payloadStart = "{ \"msgCount\": ";
+  telemetryPayload =  payloadStart + telemetrySendCount + " }";
+  telemetrySendCount++;
+
+  return (char*)telemetryPayload.c_str();
+}
+
+/*************************************/
+/*    SAS Token related functions    */
+/*************************************/
+
+/*
+ * generateMQTTPassword:
+ * The MQTT password is the generated SAS token. The process is:
+ *    1. Get the SAS token expiration time from the provided value. (Default 60 minutes).
+ *    2. Azure IoT SDK for C creates the SAS signature from this expiration time.
+ *    3. Sign and encode the SAS signature.
+ *    4. Azure IoT SDK for C creates the MQTT Password from the expiration time and the encoded,
+ *       signed SAS signature.
+ */
+static void generateMQTTPassword() 
+{
+  int result;
+
+  uint64_t sasTokenDuration = 0;
+  uint8_t signature[BUFFER_LENGTH_SAS_SIGNATURE] = {0};
+  az_span signatureAzSpan = AZ_SPAN_FROM_BUFFER(signature);
+  uint8_t encodedSignedSignature[BUFFER_LENGTH_SAS_ENCODED_SIGNED_SIGNATURE] = {0};
+  size_t encodedSignedSignatureLength = 0;
+
+  // Get the signature. It will be signed later with the decoded device key.
+  // To change the sas token duration, see IOT_CONFIG_SAS_TOKEN_EXPIRY_MINUTES in iot_configs.h
+  sasTokenDuration = getSASTokenExpirationTime(IOT_CONFIG_SAS_TOKEN_EXPIRY_MINUTES);
+  result = az_iot_hub_client_sas_get_signature(
+      &azIoTHubClient, sasTokenDuration, signatureAzSpan, &signatureAzSpan);
+  if (az_result_failed(result)) 
+  {
+    logString = "Could not get the signature for SAS Token. Return code: ";
+    LogError(logString + result);
+    exit(result);
+  }
+
+  // Sign and encode the signature (b64 encoded, HMAC-SHA256 signing).
+  // Uses the decoded device key.
+  generateSASBase64EncodedSignedSignature(
+      az_span_ptr(signatureAzSpan), az_span_size(signatureAzSpan),
+      encodedSignedSignature, sizeof(encodedSignedSignature), &encodedSignedSignatureLength);
+
+  // Get the MQTT password (SAS Token) from the base64 encoded, HMAC signed bytes.
+  az_span encodedSignedSignatureAzSpan = az_span_create(encodedSignedSignature, 
+                                                        encodedSignedSignatureLength);
+  result = az_iot_hub_client_sas_get_password(
+      &azIoTHubClient, sasTokenDuration, encodedSignedSignatureAzSpan, AZ_SPAN_EMPTY,
+      mqttPassword, sizeof(mqttPassword), NULL);
+  if (az_result_failed(result)) 
+  {
+    logString = "Could not get the MQTT password. Return code: ";
+    LogError(logString + result);
+    exit(result);

```

Comment:
Same comment for exit. The correct implementation should be getting into an infinite loop, which is then debuggable in platforms that so support it.